controllers/secret_controller.go

```diff
@@ -206,6 +217,39 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	}
 }
 
+// getUsernameAnno retrieves the username associated with a given node and ecrypts it using the given key
+func (r *SecretReconciler) getUsernameAnno(ctx context.Context, node core.Node, key []byte) (string, error) {
+	// The instance ConfigMap is the source of truth linking BYOH nodes to their underlying instances
+	instancesConfigMap := &core.ConfigMap{}
+	if err := r.client.Get(ctx, kubeTypes.NamespacedName{Namespace: r.watchNamespace,
+		Name: InstanceConfigMap}, instancesConfigMap); err != nil {
+		return "", errors.Wrap(err, "unable to get instance configmap")
+	}
+	instanceUsername, err := getNodeUsername(instancesConfigMap.Data, &node)
+	if err != nil {
+		return "", err
+	}
+	encryptedUsername, err := encryptUsername(instanceUsername, key)
+	if err != nil {
+		return "", errors.Wrapf(err, "error encrypting node %s username", node.GetName())
+	}
+	return encryptedUsername, nil
+}
+
+// applyAnnotations applies all the given annotations to the given Node resource
```

Comment:
Same comment here.

Reply:
I can see why this could be moved to the `metadata` package. I'll make this change.